analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MemberShadowsOuterStaticMember.CSharp11.cs

```diff
@@ -1,22 +1,180 @@
-﻿public interface IMath<T>
+﻿public interface OuterInterfaceWithStaticMember
 {
-    public static abstract T AbstractMethod(T t);
+    public static int StaticFoo1 => 42;
+    public static int StaticFoo2 => 42;
+    public static int StaticFoo3 => 42;
+    public static int StaticFoo4 => 42;
+    public static int StaticFoo5 => 42;
 
-    public static virtual T VirtualMethod(T t)
+    public static int StaticFoo6 => 42;
+    public static int StaticFoo7 => 42;
+
+    public abstract class InnerClass
+    {
+        public int StaticFoo1 => 42; // Noncompliant
+        public const int StaticFoo2 = 42; // Noncompliant
+        public virtual int StaticFoo3 => 42; // Noncompliant
+        public abstract int StaticFoo4 { get; } // Noncompliant
+        public static int StaticFoo5 => 42; // Noncompliant
+    }
+
+    public interface InnerInterface
+    {
+        public int StaticFoo1 => 42; // Noncompliant
+        public const int StaticFoo2 = 42; // Noncompliant
+        public virtual int StaticFoo3 => 42; // Noncompliant
+        public abstract int StaticFoo4 { get; } // Noncompliant
+        public static int StaticFoo5 => 42; // Noncompliant
+
+        public static virtual int StaticFoo6 => 42; // Compliant, you cannot invoke Interface.Member for abstract/virtual members, needs implementing type
```

Comment:
I think the comments are not 100% correct. An implementing type is not necessary, the problem is that you can invoke static virtual and static abstract interface members only on type parameters